src/services/RebalanceHLPv2Service.sol

```diff
@@ -0,0 +1,371 @@
+
+      // Taken WETH from caller and send to gmxV2WithdrawalVault for execution fee
+      weth.safeTransferFrom(msg.sender, gmxV2WithdrawalVault, _executionFee);
+      // Create a withdrawal order
+      _gmxOrderKey = gmxV2ExchangeRouter.createWithdrawal(
+        IGmxV2ExchangeRouter.CreateWithdrawalParams({
+          receiver: address(this),
+          callbackContract: address(this),
+          uiFeeReceiver: address(0),
+          market: _withdrawParam.market,
+          longTokenSwapPath: new address[](0),
+          shortTokenSwapPath: new address[](0),
+          minLongTokenAmount: _withdrawParam.minLongTokenAmount,
+          minShortTokenAmount: _withdrawParam.minShortTokenAmount,
+          shouldUnwrapNativeToken: false,
+          executionFee: _executionFee,
+          callbackGasLimit: _withdrawParam.gasLimit
+        })
+      );
+      // Update returner
+      _gmxOrderKeys[i] = _gmxOrderKey;
+      // Keep track of withdrawal history
+      withdrawalHistory[_gmxOrderKey] = _withdrawParam;
+
+      emit LogWithdrawalCreated(_gmxOrderKey, _withdrawParam);
+
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /// @notice Called by GMXv2 after a deposit execution
+  /// @param _key The key of the deposit
+  /// @param _eventData The event data emitted by GMXv2
+  function afterDepositExecution(
+    bytes32 _key,
+    IGmxV2Types.DepositProps memory /* deposit */,
+    IGmxV2Types.EventLogData memory _eventData
+  ) external onlyGmxDepositHandler {
+    // Check
+    DepositParams memory _depositParam = depositHistory[_key];
+    if (_depositParam.longToken == address(0) || _depositParam.shortToken == address(0))
+      revert IRebalanceHLPv2Service_KeyNotFound();
+
+    // Add recieved GMs as liquidity
+    uint256 _receivedGms = _eventData.uintItems.items[0].value;
+    if (_receivedGms == 0) revert IRebalanceHLPv2Service_ZeroGmReceived();
+    // GMs are sent to this contract. So we need to transfer it to vaultStorage
+    // and pull it to make sure totalAmount is updated. Then add as HLP liquidity.
+    IERC20Upgradeable(_depositParam.market).safeTransfer(address(vaultStorage), _receivedGms);
+    vaultStorage.pullToken(_depositParam.market);
+    vaultStorage.addHLPLiquidity(_depositParam.market, _receivedGms);
+
+    // Clear on hold will reduce both totalAmount and hlpLiquidityOnHold.
+    // This is to make sure that totalAmount is updated correctly.
+    // Clear on hold long token
+    if (_depositParam.longTokenAmount > 0) {
+      vaultStorage.clearOnHold(_depositParam.longToken, _depositParam.longTokenAmount);
+    }
+
+    // Clear on hold short token
+    if (_depositParam.shortTokenAmount > 0) {
+      vaultStorage.clearOnHold(_depositParam.shortToken, _depositParam.shortTokenAmount);
+    }
+
+    emit LogDepositSucceed(_key, _depositParam, _receivedGms);
+
+    // Clear deposit history
+    delete depositHistory[_key];
```

Comment:
It is named as `history`, so I'm not sure if this is an intention to delete the history.

Reply:
Just found the usage here, maybe name it `pendingDeposit/WithdrawOrder`?
<img width="468" alt="image" src="https://github.com/HMXOrg/v2-evm/assets/108852708/a38f6041-7e93-4a09-9845-ad26d4dc8665">